code/datums/components/remote_control.dm

```diff
@@ -5,53 +5,86 @@
+	///Callback to be run when user left clicks somewhere while remote controlling
+	var/datum/callback/left_click_proc
+	///Callback to be run when user right clicks somewhere while remote controlling
+	var/datum/callback/right_click_proc
 
 
 /datum/component/remote_control/Initialize(atom/movable/controlled, type)
 	. = ..()
 	if(!ismovableatom(controlled))
 		return COMPONENT_INCOMPATIBLE
 	src.controlled = controlled
-	update_clickproc(null, type)
-	RegisterSignal(controlled, COMSIG_UNMANNED_TURRET_UPDATED, .proc/update_clickproc)
+	update_left_clickproc(null, type)
+	RegisterSignal(controlled, COMSIG_UNMANNED_TURRET_UPDATED, .proc/update_left_clickproc)
+	RegisterSignal(controlled, COMSIG_UNMANNED_ABILITY_UPDATED, .proc/update_right_clickproc)
 	RegisterSignal(parent, COMSIG_REMOTECONTROL_TOGGLE, .proc/toggle_remote_control)
-	RegisterSignal(controlled, list(COMSIG_PARENT_QDELETING, COMSIG_REMOTECONTROL_UNLINK), .proc/on_control_terminate)
+	RegisterSignal(controlled, COMSIG_PARENT_QDELETING, .proc/on_control_terminate)
+	RegisterSignal(controlled, COMSIG_MOVABLE_HEAR, .proc/on_hear)
+	RegisterSignal(parent, list(COMSIG_REMOTECONTROL_UNLINK, COMSIG_PARENT_QDELETING), .proc/on_control_terminate)
+
 
 /datum/component/remote_control/Destroy(force=FALSE, silent=FALSE)
 	if(is_controlling)
 		remote_control_off()
 	controlled = null
-	click_proc = null
+	left_click_proc = null
+	right_click_proc = null
 	return ..()
 
 ///Called when Controlling should not be resumed and deleted the component
 /datum/component/remote_control/proc/on_control_terminate(datum/source)
 	SIGNAL_HANDLER
 	qdel(src)
 
+///Called when the controlled atom hear a sound
+/datum/component/remote_control/proc/on_hear(datum/source, message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, message_mode)
+	SIGNAL_HANDLER
+	user?.Hear(message, speaker, message_language, raw_message, radio_freq, spans, message_mode)
+
+/// Called when the remote controller wants to speak through the controlled atom
+/datum/component/remote_control/proc/on_relayed_speech(datum/source, message, language)
+	SIGNAL_HANDLER
+	message = trim(message)
+	if(!message)
+		return
+	controlled.say(message, language = language)
+	return COMSIG_RELAYED_SPEECH_DEALT
+
 
 ///Updates the clickproc to a passed type of turret
-/datum/component/remote_control/proc/update_clickproc(datum/source, type)
+/datum/component/remote_control/proc/update_left_clickproc(datum/source, type)
 	SIGNAL_HANDLER
 	switch(type)
-		if(TURRET_TYPE_HEAVY, TURRET_TYPE_LIGHT, TURRET_TYPE_DROIDLASER)
-			click_proc = CALLBACK(src, .proc/uv_handle_click)
-		if(TURRET_TYPE_EXPLOSIVE)
-			click_proc = CALLBACK(src, .proc/uv_handle_click_explosive)
+		if(/obj/item/uav_turret, /obj/item/uav_turret/heavy, /obj/item/uav_turret/droid)
+			left_click_proc = CALLBACK(src, .proc/uv_handle_click)
+		if(/obj/item/explosive/plastique)
+			left_click_proc = CALLBACK(src, .proc/uv_handle_click_explosive)
```

Comment:
why

Reply:
This is your code?

Reply:
Why did you change it to a strict type check from a var check

Reply:
Change it back strict type chacks like this are bad